src/routes/concepts/stores.mdx

```diff
@@ -238,157 +232,145 @@ setStore("users", (otherUsers) => [
 
 ```jsx
-import { produce } from "solid-js/store"
+import { produce } from "solid-js/store";
 
 // without produce
-setStore("users", 0, "username", "newUsername")
-setStore("users", 0, "location", "newLocation")
+setStore(
+	"users",
+	0,
+	(user) => ({
+		username: "newUsername";
+		location: "newLocation";
+}));
 
 // with produce
 setStore(
 	"users",
 	0,
 	produce((user) => {
-		user.username = "newUsername"
-		user.location = "newLocation"
+		user.username = "newUsername";
+		user.location = "newLocation";
 	})
-)
+);
 ```
 
-`produce` and `setStore` do have distinct functionalities.
-While both can be used to modify the state, the key distinction lies in how they handle data.
-`produce` allows you to work with a temporary draft of the state, apply the changes, then produce a new [immutable](https://developer.mozilla.org/en-US/docs/Glossary/Immutable) version of the store.
-Comparatively, `setStore` provides a more straightforward way to update the store directly, without creating a new version.
-
-It's important to note, however, `produce` is specifically designed to work with **arrays** and **objects**.
-Other collection types, such as JavaScript [Sets](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Set) and [Maps](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map), are not compatible with this utility.
+`produce` and `setStore` both modify the store, but the key distinction is how they handle data.
+`produce` gives you a temporary draft of the state to update, then produces a new [immutable](https://developer.mozilla.org/en-US/docs/Glossary/Immutable) version of the store with the updated values.
+`setStore` updates the store directly, without creating a new version.
 
-### Data integration with `reconcile`
+### Integrating data with `reconcile`
 
-When new information needs to be merged into an existing store `reconcile` can be useful.
-`reconcile` will determine the differences between new and existing data and initiate updates only when there are _changed_ values, thereby avoiding unnecessary updates.
+The `reconcile` utility can be useful when you need to merge new data into a store.
+It determines the differences between the existing data and the incoming data and modifies only the _changed_ values, avoiding unnecessary updates.
+In this example, only `'koala'`, the new addition, will cause an update:
 
 ```jsx
 const { createStore, reconcile } from "solid-js/stores"
 
-const [data, setData] = createStore({
+const [store, setStore] = createStore({
 	animals: ['cat', 'dog', 'bird', 'gorilla']
 })
 
-const newData = getNewData() // eg. contains ['cat', 'dog', 'bird', 'gorilla', 'koala']
-setData('animals', reconcile(newData))
-
+const newData = {
+	animals: ['cat', 'dog', 'bird', 'gorilla', 'koala']
+}
+setStore('animals', reconcile(newData))
```

Comment:
I know this example is bad because the DOM behaves exactly the same way whether you use reconcile or not in this case. Only `koala` is added and no other nodes rerender whether you use reconcile or not. You can watch the DOM tree in the devtools on the playground site for confirmation, see https://playground.solidjs.com/anonymous/c37c2d5d-d820-4507-859c-5a19ca73edbf

But I don't know enough to come up with an example that would actually demonstrate the difference.

Reply:
@SpencerWhitehead7 I modified your example a little to make it actually show the difference. Since solid compares the values it doesn't re-execute the original items when switching to the new string array because the values match, but by switching to objects which will have different refs it will re-execute unless you use reconcile. You have to use a console log to see that it re-executes because the result is the same either way https://playground.solidjs.com/anonymous/4ee128da-0098-4b3f-84ea-da447f94b81e

Reply:
Ah, thank you so much, I checked it out and I can see it working.

If this PR/issue is ever reopened I will definitely update it with your example.